src/server/services/wmts/qgswmtsgetcapabilities.cpp

```diff
@@ -0,0 +1,571 @@
+
+    //ows:DCP
+    QDomElement dcpElement = doc.createElement( QStringLiteral( "ows:DCP" )/*ows:DCP*/ );
+    getCapabilitiesElement.appendChild( dcpElement );
+    QDomElement httpElement = doc.createElement( QStringLiteral( "ows:HTTP" )/*ows:HTTP*/ );
+    dcpElement.appendChild( httpElement );
+
+    //Prepare url
+    QString hrefString = serviceUrl( request, project );
+
+    //ows:Get
+    QDomElement getElement = doc.createElement( QStringLiteral( "ows:Get" )/*ows:Get*/ );
+    getElement.setAttribute( QStringLiteral( "xlink:href" ), hrefString );
+    httpElement.appendChild( getElement );
+
+    //ows:Operation element with name GetTile
+    QDomElement getTileElement = getCapabilitiesElement.cloneNode().toElement();//this is the same as 'GetCapabilities'
+    getTileElement.setAttribute( QStringLiteral( "name" ), QStringLiteral( "GetTile" ) );
+    operationsMetadataElement.appendChild( getTileElement );
+
+    //ows:Operation element with name GetFeatureInfo
+    QDomElement getFeatureInfoElement = getCapabilitiesElement.cloneNode().toElement();//this is the same as 'GetCapabilities'
+    getFeatureInfoElement.setAttribute( QStringLiteral( "name" ), QStringLiteral( "GetFeatureInfo" ) );
+    operationsMetadataElement.appendChild( getFeatureInfoElement );
+
+    // End
+    return operationsMetadataElement;
+  }
+
+  QDomElement getContentsElement( QDomDocument &doc, QgsServerInterface *serverIface, const QgsProject *project )
+  {
+    /*
+     * Adding layer list in ContentMetadata
+     */
+    QDomElement contentsElement = doc.createElement( QStringLiteral( "Contents" )/*wmts:Contents*/ );
+
+    QList< tileMatrixSetDef > tmsList = getTileMatrixSetList( project );
+    if ( !tmsList.isEmpty() )
+    {
+      // get layer list
+      QList< layerDef > wmtsLayers = getWmtsLayerList( serverIface, project );
+      if ( !wmtsLayers.isEmpty() )
+      {
+        appendLayerElements( doc, contentsElement, wmtsLayers, tmsList, project );
+      }
+
+      appendTileMatrixSetElements( doc, contentsElement, tmsList );
+    }
+
+    //End
+    return contentsElement;
+  }
+  namespace
+  {
+    void appendLayerElements( QDomDocument &doc, QDomElement &contentsElement,
+                              QList< layerDef > wmtsLayers, QList< tileMatrixSetDef > tmsList,
+                              const QgsProject *project )
+    {
+      QgsCoordinateReferenceSystem wgs84 = QgsCoordinateReferenceSystem::fromOgcWmsCrs( GEO_EPSG_CRS_AUTHID );
+      // Define InfoFormat helper
+      std::function < void ( QDomElement &, const QString & ) > appendInfoFormat = [&doc]( QDomElement & elem, const QString & format )
+      {
+        QDomElement formatElem = doc.createElement( QStringLiteral( "InfoFormat" )/*wmts:InfoFormat*/ );
+        formatElem.appendChild( doc.createTextNode( format ) );
+        elem.appendChild( formatElem );
+      };
+
+      for ( const layerDef wmtsLayer : wmtsLayers )
```

Comment:
`&` is missing to have a proper const reference. 

Can you take a look to other range-based for loops in this file too? Because it seems to be missing for many of them.